src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -326,6 +327,10 @@ private static bool CompileProject(ProjectContext context, string configuration,
                 return false;
             }
 
+            var assemblyInfo = Path.Combine(intermediateOutputPath, $"dotnet-compile.assemblyinfo.cs");
```

Comment:
As discussed in person, move this to `dotnet-compile-csc`. Have `dotnet-compile` generate the data and pass it along in the `rsp` file via new command line switches (i.e. `project-version` or something)